src/protocols/secure_channel/tests/TestCASESession.cpp

```diff
@@ -432,6 +522,40 @@ void CASE_SecurePairingSerializeTest(nlTestSuite * inSuite, void * inContext)
     chip::Platform::Delete(testPairingSession2);
 }
 
+void CASE_SecurePairingEmptySigmaR1(nlTestSuite * inSuite, void * inContext)
+{
+    TestCASESecurePairingDelegate delegateCommissioner;
+    TestCASESessionIPK pairingCommissioner;
+
+    class R1Emptier : public CASETestTransport::Adjuster
+    {
+    public:
+        R1Emptier(CASETestTransport & transport) : CASETestTransport::Adjuster(transport) {}
+
+    private:
+        CHIP_ERROR AdjustMessage(const PayloadHeader & payloadHeader, System::PacketBufferHandle & msg) override
+        {
+            if (payloadHeader.HasMessageType(SecureChannel::MsgType::CASE_SigmaR1))
+            {
+                // Empty out the payload
+                msg->SetDataLength(0);
+            }
+            return CHIP_NO_ERROR;
+        }
```

Comment:
This breaks the message, which helps test some more cases. But what about testing for different contents of the payload (e.g. not assuming that the sender of a message is doing things right, to test attack/fuzz situations)? How do we make test cases to validate sender data contents, and receiver parsing behavior?

Reply:
With this setup, you basically need a different adjuster class (or perhaps an adjuster class parametrized on some data) per test you want to run, yes...